src/kvstore/kvstore_local.h

```diff
@@ -256,7 +256,13 @@ class KVStoreLocal : public KVStore {
     auto validator = [this](const int key, const NDArray& nd) -> bool {
       auto stype = nd.storage_type();
       // valid NDArray
-      if (stype == kDefaultStorage || stype == kRowSparseStorage) return true;
+      if (stype == kDefaultStorage || stype == kRowSparseStorage
```

Comment:
Injecting macro into if statement is kind of hard to read. Is it better if we change to:
```
bool valid_stype = stype == kDefaultStorage || stype == kRowSparseStorage;
#if MXNET_USE_MKLDNN == 1
     valid_stype = valid_stype  || stype == kMKLDNNStorage
#endif
if (valid_stype) return true;

```

Reply:
right. I have updated it.